Switch to audio then back to video crash

### Steps to reproduce

Open a video, turn on audio mode, turn on video mode again

### Expected behavior

App isn't crashing

### Actual behavior

App is crashing

### LibreTube version

0.23.0

### Android version

Android 14

### Other details

java.lang.RuntimeException: Unable to stop service com.github.libretube.services.OnlinePlayerService@b6752b0: java.lang.IllegalArgumentException: Receiver not registered: androidx.media3.common.util.NetworkTypeObserver$Receiver@a500529
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5111)
	at android.app.ActivityThread.-$$Nest$mhandleStopService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2405)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8532)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:359)
Caused by: java.lang.IllegalArgumentException: Receiver not registered: androidx.media3.common.util.NetworkTypeObserver$Receiver@a500529
	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1670)
	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1905)
	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:821)
	at com.github.libretube.services.OnlinePlayerService.onDestroy(SourceFile:34)
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5091)
	... 11 more
    

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

I also encountered this error:

```
java.lang.RuntimeException: Unable to stop service com.github.libretube.services.OnlinePlayerService@218a77b: java.lang.IllegalArgumentException: Receiver not registered: com.github.libretube.services.OnlinePlayerService$playerActionReceiver$1@ba8f998
	at android.app.ActivityThread.handleStopService(ActivityThread.java:4706)
	at android.app.ActivityThread.-$$Nest$mhandleStopService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2185)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7924)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.lang.IllegalArgumentException: Receiver not registered: com.github.libretube.services.OnlinePlayerService$playerActionReceiver$1@ba8f998
	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1662)
	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1836)
	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:812)
	at com.github.libretube.services.OnlinePlayerService.onDestroy(OnlinePlayerService.kt:435)
	at android.app.ActivityThread.handleStopService(ActivityThread.java:4686)
	... 9 more
    
```

I also encounter this error
java.lang.RuntimeException: Unable to stop service com.github.libretube.services.OnlinePlayerService@589563c: java.lang.IllegalArgumentException: Receiver not registered: androidx.media3.common.util.NetworkTypeObserver$Receiver@2282fc5
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5345)
	at android.app.ActivityThread.access$2400(ActivityThread.java:324)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2379)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8855)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
Caused by: java.lang.IllegalArgumentException: Receiver not registered: androidx.media3.common.util.NetworkTypeObserver$Receiver@2282fc5
	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1656)
	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1867)
	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:788)
	at com.github.libretube.services.OnlinePlayerService.onDestroy(SourceFile:34)
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5325)
	... 9 more

--- a/app/src/main/java/com/github/libretube/services/OfflinePlayerService.kt
+++ b/app/src/main/java/com/github/libretube/services/OfflinePlayerService.kt
@@ -4,9 +4,7 @@ import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
 import android.content.IntentFilter
-import android.os.Handler
 import android.os.IBinder
-import android.os.Looper
 import androidx.annotation.OptIn
 import androidx.core.app.NotificationCompat
 import androidx.core.app.ServiceCompat
@@ -43,8 +41,6 @@ import kotlinx.coroutines.withContext
  * A service to play downloaded audio in the background
  */
 class OfflinePlayerService : LifecycleService() {
-    val handler = Handler(Looper.getMainLooper())
-
     private var player: ExoPlayer? = null
     private var nowPlayingNotification: NowPlayingNotification? = null
     private lateinit var videoId: String
@@ -209,14 +205,18 @@ class OfflinePlayerService : LifecycleService() {
         saveWatchPosition()
 
         nowPlayingNotification?.destroySelf()
-
-        player?.stop()
-        player?.release()
-        player = null
         nowPlayingNotification = null
-
         watchPositionTimer.destroy()
-        unregisterReceiver(playerActionReceiver)
+
+        runCatching {
+            player?.stop()
+            player?.release()
+        }
+        player = null
+
+        runCatching {
+            unregisterReceiver(playerActionReceiver)
+        }
 
         ServiceCompat.stopForeground(this, ServiceCompat.STOP_FOREGROUND_REMOVE)
         stopSelf()

--- a/app/src/main/java/com/github/libretube/services/OnlinePlayerService.kt
+++ b/app/src/main/java/com/github/libretube/services/OnlinePlayerService.kt
@@ -31,7 +31,6 @@ import com.github.libretube.api.RetrofitInstance
 import com.github.libretube.api.obj.Segment
 import com.github.libretube.api.obj.Streams
 import com.github.libretube.constants.IntentData
-import com.github.libretube.constants.PreferenceKeys
 import com.github.libretube.db.DatabaseHelper
 import com.github.libretube.enums.NotificationId
 import com.github.libretube.enums.PlayerEvent
@@ -42,7 +41,6 @@ import com.github.libretube.extensions.toID
 import com.github.libretube.extensions.updateParameters
 import com.github.libretube.helpers.PlayerHelper
 import com.github.libretube.helpers.PlayerHelper.checkForSegments
-import com.github.libretube.helpers.PreferenceHelper
 import com.github.libretube.helpers.ProxyHelper
 import com.github.libretube.obj.PlayerNotificationData
 import com.github.libretube.parcelable.PlayerData
@@ -372,7 +370,7 @@ class OnlinePlayerService : LifecycleService() {
 
         val (uri, mimeType) =
             if (!PlayerHelper.useHlsOverDash && streams.audioStreams.isNotEmpty() && !PlayerHelper.disablePipedProxy) {
-                PlayerHelper.createDashSource(streams, this,) to MimeTypes.APPLICATION_MPD
+                PlayerHelper.createDashSource(streams, this) to MimeTypes.APPLICATION_MPD
             } else {
                 ProxyHelper.unwrapStreamUrl(streams.hls.orEmpty())
                     .toUri() to MimeTypes.APPLICATION_M3U8
@@ -427,12 +425,17 @@ class OnlinePlayerService : LifecycleService() {
         PlayingQueue.resetToDefaults()
 
         if (this::nowPlayingNotification.isInitialized) nowPlayingNotification.destroySelf()
-
-        player?.stop()
-        player?.release()
-
         watchPositionTimer.destroy()
-        unregisterReceiver(playerActionReceiver)
+        handler.removeCallbacksAndMessages(null)
+
+        runCatching {
+            player?.stop()
+            player?.release()
+        }
+
+        runCatching {
+            unregisterReceiver(playerActionReceiver)
+        }
 
         // called when the user pressed stop in the notification
         // stop the service from being in the foreground and remove the notification

--- a/app/src/main/java/com/github/libretube/ui/activities/OfflinePlayerActivity.kt
+++ b/app/src/main/java/com/github/libretube/ui/activities/OfflinePlayerActivity.kt
@@ -300,12 +300,19 @@ class OfflinePlayerActivity : BaseActivity() {
     override fun onDestroy() {
         saveWatchPosition()
 
-        playerViewModel.player = null
-        player.release()
-        watchPositionTimer.destroy()
         nowPlayingNotification?.destroySelf()
+        nowPlayingNotification = null
+        watchPositionTimer.destroy()
 
-        unregisterReceiver(playerActionReceiver)
+        playerViewModel.player = null
+        runCatching {
+            player.stop()
+            player.release()
+        }
+
+        runCatching {
+            unregisterReceiver(playerActionReceiver)
+        }
 
         super.onDestroy()
     }

--- a/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
+++ b/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
@@ -783,17 +783,30 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
     override fun onDestroy() {
         super.onDestroy()
 
+        saveWatchPosition()
+
+        viewModel.nowPlayingNotification?.destroySelf()
+        viewModel.nowPlayingNotification = null
+        watchPositionTimer.destroy()
+        handler.removeCallbacksAndMessages(null)
+
         if (this::exoPlayer.isInitialized) {
             exoPlayer.removeListener(playerListener)
-
-            // the player could also be a different instance because a new player fragment
-            // got created in the meanwhile
-            if (!viewModel.isOrientationChangeInProgress && viewModel.player == exoPlayer) {
-                viewModel.player = null
-                viewModel.trackSelector = null
-            }
-
             exoPlayer.pause()
+
+            runCatching {
+                // the player could also be a different instance because a new player fragment
+                // got created in the meanwhile
+                if (!viewModel.isOrientationChangeInProgress) {
+                    if (viewModel.player == exoPlayer) {
+                        viewModel.player = null
+                        viewModel.trackSelector = null
+                    }
+
+                    exoPlayer.stop()
+                    exoPlayer.release()
+                }
+            }
 
             if (PlayerHelper.pipEnabled) {
                 // disable the auto PiP mode for SDK >= 32
@@ -802,35 +815,29 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
             }
         }
 
-        handler.removeCallbacksAndMessages(null)
-
         runCatching {
             // unregister the receiver for player actions
             context?.unregisterReceiver(playerActionReceiver)
         }
 
+        // restore the orientation that's used by the main activity
+        (context as MainActivity).requestOrientationChange()
+
         _binding = null
-
-        stopVideoPlay()
-
-        watchPositionTimer.destroy()
     }
 
-    private fun stopVideoPlay() {
-        try {
-            saveWatchPosition()
+    private fun killPlayerFragment() {
+        viewModel.isFullscreen.value = false
+        viewModel.isMiniPlayerVisible.value = false
 
-            viewModel.nowPlayingNotification?.destroySelf()
-            viewModel.nowPlayingNotification = null
+        // dismiss the fullscreen dialog if it's currently visible
+        // otherwise it would stay alive while being detached from this fragment
+        fullscreenDialog.dismiss()
+        binding.player.currentWindow = null
 
-            if (!viewModel.isOrientationChangeInProgress) {
-                exoPlayer.stop()
-                exoPlayer.release()
-            }
-
-            (context as MainActivity).requestOrientationChange()
-        } catch (e: Exception) {
-            e.printStackTrace()
+        binding.playerMotionLayout.transitionToEnd()
+        mainActivity.supportFragmentManager.commit {
+            remove(this@PlayerFragment)
         }
     }
 
@@ -1510,22 +1517,6 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
     private fun shouldStartPiP(): Boolean {
         return shouldUsePip() && exoPlayer.isPlaying &&
                 !BackgroundHelper.isBackgroundServiceRunning(requireContext())
-    }
-
-    private fun killPlayerFragment() {
-        stopVideoPlay()
-        viewModel.isFullscreen.value = false
-        viewModel.isMiniPlayerVisible.value = false
-
-        // dismiss the fullscreen dialog if it's currently visible
-        // otherwise it would stay alive while being detached from this fragment
-        fullscreenDialog.dismiss()
-        binding.player.currentWindow = null
-
-        binding.playerMotionLayout.transitionToEnd()
-        mainActivity.supportFragmentManager.commit {
-            remove(this@PlayerFragment)
-        }
     }
 
     /**

